homeassistant/components/iron_os/coordinator.py

```diff
@@ -147,3 +150,21 @@ async def _async_update_data(self) -> SettingsDataResponse:
                 _LOGGER.debug("Failed to fetch settings", exc_info=e)
 
         return self.data or SettingsDataResponse()
+
+    async def write(self, characteristic: CharSetting, value: bool) -> None:
+        """Write value to the settings characteristic."""
+
+        try:
+            await self.device.write(characteristic, value)
+        except CommunicationError as e:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="submit_setting_failed",
+            ) from e
+        else:
+            # prevent switch bouncing while waiting for coordinator to finish refresh
+            self.data.update(
+                cast(SettingsDataResponse, {characteristic.name.lower(): value})
+            )
+            self.async_update_listeners()
+            await self.async_request_refresh()
```

Comment:
```suggestion
        # prevent switch bouncing while waiting for coordinator to finish refresh
        self.data.update(
            cast(SettingsDataResponse, {characteristic.name.lower(): value})
        )
        self.async_update_listeners()
        await self.async_request_refresh()
```
just saving some indent